src/api/utils/encryption.py

```
@@ -0,0 +1,73 @@
+        super().__init__(length, **kwargs)
+        self.encryption_service = EncryptionService()
+
+    def process_bind_param(self, value, dialect):
+        """Encrypt value before storing in database"""
+        if value is None:
+            return None
+        try:
+            encrypted = self.encryption_service.encrypt(value)
+            return encrypted.decode('utf-8')  # Store as string in DB
+        except Exception as e:
+            raise ValueError(f"Error encrypting value: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)